sshd-core/src/main/java/org/apache/sshd/common/util/io/IoUtils.java

```diff
@@ -257,6 +260,56 @@ public static void setPermissionsToFile(File f, Collection<PosixFilePermission>
     }
 
     /**
+     * <P>Get file owner.</P>
+     *
+     * @param path  The {@link Path}
+     * @param options The {@link LinkOption}s to use when querying the owner
+     * @return Owner of the file or null if unsupported
+     * @throws IOException If failed to access the file system
+     */
+    public static String getFileOwner(Path path, LinkOption... options) throws IOException {
+        String owner = null;
+        try {
+            UserPrincipal principal = Files.getOwner(path, options);
+            if (principal != null) {
+                owner = principal.getName();
+            }
+        } catch (UnsupportedOperationException e) {
+            // ignore
+        }
+        return owner;
+    }
+
+    /**
+     * <P>Get current user by file ownership.</P>
+     *
+     * @return Owner by creating a file and check ownership
+     * @throws IOException If failed to access the file system
+     */
+    public static String getCurrentUser() throws IOException {
+        synchronized (CURRENT_USER_HOLDER) {
+            String owner = CURRENT_USER_HOLDER.get();
+            if (owner == null) {
+                File t = null;
+                try {
+                    t = File.createTempFile("ssh-", ".tmp");
+                    owner = getFileOwner(t.toPath());
+                    if (owner == null) {
+                        // mark as attempted
+                        owner = "";
+                    }
+                } finally {
+                    if (t != null) {
+                        t.delete();
+                    }
+                }
+                CURRENT_USER_HOLDER.set(owner);
+            }
```

Comment:
The let me suggest a compromise:
```java
/**
  * Property that can be used to override the reported value from {@link #getCurrentUser()}
  * if not set then &quot;user.name&quot; system property is used
  */
public static final String CURRENT_USER_OVERRIDE_PROP = "org.apache.sshd.currentUser";
private static final AtomicReference<String> CURRENT_USER_HOLDER = new ...(null);

public static String getCurrentUser() {
    String username = null;
    synchronized (CURRENT_USER_HOLDER) {
        username = CURRENT_USER_HOLDER.get();
        if (username == null) {  // have we already resolved it ?
           return username;
       }

       username = System.getProperty(CURRENT_USER_OVERRIDE_PROP, System.getProperty("user.name"));
       ??? ValidateUtils.checkNotEmpty(username, "No username available"); ???
       CURRENT_USER_HOLDER.set(username);
    }

    return username;
}

// can be used to programmatically set the username reported by {@link #getCurrentUser()}
public static void setCurrentUser(String username) {
    synchronized (CURRENT_USER_HOLDER) {    // not really necessary, but let's be consistent
          CURRENT_USER_HOLDER.set(username);
    }
```

If we agree on this, then you need locate *all* the other places in the code (and there are very few) where "user.name" is used and replace it with "IoUtils.getCurrentUser()". Also, thinking about it, the "get/setCurrentUser" methods should reside in _OsUtils_